robotools/evotools/__init__.py

```diff
@@ -181,6 +181,369 @@ def _int_to_tip(tip_int: int):
+        Volume in microliters (will be rounded to 2 decimal places); if several tips are used, these tips may aspirate individual volumes -> use list in these cases
+    liquid_class : str, optional
+        Overwrites the liquid class for this step (max 32 characters)
+    tips : list of int
+        Tip(s) that will be selected (out of tips 1-8)
+    """
+    # required parameters
+    if labware is None:
+        raise ValueError("Missing required paramter: labware")
+    if not isinstance(labware, liquidhandling.Labware):
+        raise ValueError(f"Invalid labware: {labware}")
+
+    if labware_position is None:
+        raise ValueError("Missing required paramter: position")
+    if not all(isinstance(position, int) for position in labware_position) or any(
+        position < 0 for position in labware_position
+    ):
+        raise ValueError(f"Invalid position: {labware_position}")
+
+    if volume is None:
+        raise ValueError("Missing required parameter: volume")
+    if type(volume) == list:
+        for vol in volume:
+            try:
+                vol = float(vol)
+            except:
+                raise ValueError(f"Invalid volume: {vol}")
+            if vol < 0 or vol > 7158278 or numpy.isnan(vol):
+                raise ValueError(f"Invalid volume: {vol}")
+            if max_volume is not None and vol > max_volume:
+                raise InvalidOperationError(f"Volume of {vol} exceeds max_volume.")
+            if not len(vol) == len(tips):
+                raise Exception(
+                    f"Tips and volume lists have different lengths ({len(tips)} and {len(vol)}, respectively)."
+                )
+    else:
+        try:
+            volume = float(volume)
+        except:
+            raise ValueError(f"Invalid volume: {volume}")
+        if volume < 0 or volume > 7158278 or numpy.isnan(volume):
+            raise ValueError(f"Invalid volume: {volume}")
+        if max_volume is not None and volume > max_volume:
+            raise InvalidOperationError(f"Volume of {volume} exceeds max_volume.")
+
+    # optional parameters
+    if not isinstance(liquid_class, str) or ";" in liquid_class:
+        raise ValueError(f"Invalid liquid_class: {liquid_class}")
+
+    def _int_to_tip(tip_int: int):
+        """Asserts a Tecan Tip class to an int between 1 and 8."""
+        if not 1 <= tip_int <= 8:
+            raise ValueError(
+                f"Tip is {tip} with type {type(tip)}, but should be an int between 1 and 8 for _int_to_tip conversion."
+            )
+        if tip_int == 1:
+            return Tip.T1
+        elif tip_int == 2:
+            return Tip.T2
+        elif tip_int == 3:
+            return Tip.T3
+        elif tip_int == 4:
+            return Tip.T4
+        elif tip_int == 5:
+            return Tip.T5
+        elif tip_int == 6:
+            return Tip.T6
+        elif tip_int == 7:
+            return Tip.T7
+        elif tip_int == 8:
+            return Tip.T8
```

Comment:
This function can be extracted, and it should raise an exception in the final `else` branch.